### PR TITLE
function process

### PR DESCRIPTION
**Pull request by tx19980520**
1. support code prepare
2. support run different environment code
3. implement golang wrapper
4. add test for instance and init cmd

### CODE REVIEW

**Comment by tx19980520** [comment]
root@tass:/home/ubuntu/scheduler# go test -p 1 ./...
?       github.com/tass-io/scheduler    [no test files]
?       github.com/tass-io/scheduler/cmd        [no test files]
?       github.com/tass-io/scheduler/pkg/dto    [no test files]
?       github.com/tass-io/scheduler/pkg/env    [no test files]
?       github.com/tass-io/scheduler/pkg/event  [no test files]
ok      github.com/tass-io/scheduler/pkg/event/schedule 5.027s
?       github.com/tass-io/scheduler/pkg/http   [no test files]
?       github.com/tass-io/scheduler/pkg/http/controller        [no test files]
ok      github.com/tass-io/scheduler/pkg/initial        2.206s
?       github.com/tass-io/scheduler/pkg/initial/wrapper        [no test files]
?       github.com/tass-io/scheduler/pkg/middleware     [no test files]
ok      github.com/tass-io/scheduler/pkg/middleware/lsds        0.040s
?       github.com/tass-io/scheduler/pkg/runner [no test files]
ok      github.com/tass-io/scheduler/pkg/runner/fnscheduler     4.026s
?       github.com/tass-io/scheduler/pkg/runner/helper  [no test files]
ok      github.com/tass-io/scheduler/pkg/runner/instance        4.314s
ok      github.com/tass-io/scheduler/pkg/runner/lsds    5.045s
?       github.com/tass-io/scheduler/pkg/schedule       [no test files]
?       github.com/tass-io/scheduler/pkg/span   [no test files]
?       github.com/tass-io/scheduler/pkg/store  [no test files]
?       github.com/tass-io/scheduler/pkg/tools/base64   [no test files]
?       github.com/tass-io/scheduler/pkg/tools/common   [no test files]
?       github.com/tass-io/scheduler/pkg/tools/errorutils       [no test files]
?       github.com/tass-io/scheduler/pkg/tools/k8sutils [no test files]
?       github.com/tass-io/scheduler/pkg/tools/log      [no test files]
ok      github.com/tass-io/scheduler/pkg/workflow       4.041s
?       github.com/tass-io/scheduler/test       [no test files]
ok      github.com/tass-io/scheduler/test/initial       5.516s
ok      github.com/tass-io/scheduler/test/pipe  3.856s
ok      github.com/tass-io/scheduler/test/switch        2.860s

File: pkg/runner/instance/conn.go [hunk 1]
```diff
@@ -29,64 +31,70 @@ var (
...
+type FunctionResponse struct {
 	Id     string
 	Result map[string]interface{}
 }
 
-type producer struct {
+type Producer struct {
 	f              *os.File
-	requestChannel chan functionRequest
+	demo           interface{}
```
Comment (linxuyalun) [inline pkg/runner/instance/conn.go:60]: What's this? data?

Comment (tx19980520) [reply pkg/runner/instance/conn.go:60]: the demo just given the `Type`.
This change was made to be more general

File: pkg/initial/init.go [hunk 2]
```diff
@@ -82,9 +92,50 @@ func codePrepareAndExec(code string) {
...
+				os.Exit(2)
+			}
+			if err := syscall.Exec(binary, []string{entryPath}, os.Environ()); err != nil {
+				zap.S().Errorw("init exec error", "err", err)
+				os.Exit(2)
+			}
+		}
+	case "Golang":
+		{
+			entryPath := codePath + "/main"
```
[the remark this answers is not in the document]
Comment (tx19980520) [reply pkg/initial/init.go:122]: assume code is injected and build

File: pkg/env/env.go [hunk 3]
```diff
@@ -1,11 +1,17 @@
 package env
 
 const (
-	LSDSWait = "LSDSWait"
-	Policy = "policy"
-	Local = "local"
+	LSDSWait                = "LSDSWait"
```
[the remark this answers is not in the document]
Comment (tx19980520) [reply pkg/env/env.go]: because it is waiting for merge

**Review by linxuyalun — APPROVED** [review]
LGTM

**Pull request merged**